### DISCUSSION TITLE
en/chapter_preface/suggestions/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#1387] [Announcements]
# en/chapter_preface/suggestions/

Data Structures and Algorithms Crash Course with Animated Illustrations and Off-the-Shelf Code

https://www.hello-algo.com/en/chapter_preface/suggestions/

**Comment by bit-scientist** [comment]
So when it said " book mainly covers “Stage 1,” aiming to help you more efficiently embark on Stages 2 and 3.", does it mean this book also covers stage 2 and stage 3? Or you expect learners to find alternative sources somewhere else?

**Reply by krahets** [reply]
Hi! This book's goal is to build a solid foundation so you can move on to Stages 2 and 3 more efficiently, but those stages are not covered in depth in this book. You’re expected to explore them with other resources (and practice).

**Comment by harichselvamc** [comment]
Which branch should we clone for learning? Could you please mention that as well?
